src/components/Settings/General.vue

```diff
@@ -1,56 +1,155 @@
+            <div v-tooltip="$t(`tooltips.edit-experience`)" class="icon-wrapper hover:text-primary"
+              @click="editDialogToggle(slotProps.item)">
+              <i class="pi pi-pencil"></i>
+            </div>
+          </div>
+        </div>
+      </template>
+    </OrderList>
+    <ExperiencePopup 
+      v-if="editModalIsOpen" 
+      :payload="editModalPayload"  
+      :type="type" 
+      :update-experince-info="updateExperinceInfo"
+      @close="editClosemodal" 
+      />
   </div>
 </template>
 <script>
-  import InputSwitch from "primevue/inputswitch"
-  import { getModule } from "vuex-module-decorators"
-  import Settings from "@/store/Modules/Settings"
-  const settingsModule = getModule(Settings)
-  export default {
-    name: "General",
-    components: { InputSwitch },
-    computed: {
-      experimentalFeatures() {
-        return settingsModule.data.experimentalFeatures
-      },
+import InputSwitch from "primevue/inputswitch"
+import { getModule } from "vuex-module-decorators"
+import Settings from "@/store/Modules/Settings"
+import Button from "primevue/button"
+import Dialog from "primevue/dialog"
+import { nanoid } from "nanoid"
+import OrderList from 'primevue/orderlist';
+
+import { errorToast, successToast } from "@/utils/toastService"
+import ExperiencePopup from "@/components/Dialog/ExperiencePopup.vue"
+const settingsModule = getModule(Settings)
+export default {
+  name: "General",
+  components: { InputSwitch, OrderList, Button, ExperiencePopup },
+  data() {
+    return {
+      listOfExperiences: [],
+      updatedExperienceList: [],
+      label: "",
+      icon: "",
+      url: "",
+      editModalPayload: {},
+      editModalIsOpen: false,
+      type: "",
+      saveDiabale: true
+    }
+  },
+
+  computed: {
+    editDialog() {
+      return settingsModule.data.editDialog
     },
-    methods: {
-      handleFeatureToggle(enabled, id) {
-        settingsModule.toggleExperimentalFeatures({ id, enabled })
-      },
+  },
+  mounted() {
+    this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
+    console.log("listOfExperiences", this.listOfExperiences)
+  },
+  methods: {
+    menuList(orderList, event) {
+      console.log("list", orderList, event)
+      console.log("list1", event)
+      this.saveDiabale = false
+      this.updatedExperienceList = orderList
+      // this.updatedExperienceList = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
     },
-  }
+    saveOrderList() {
+      settingsModule.updateList(this.updatedExperienceList)
+      this.saveDiabale = true
+      this.listOfExperiences = JSON.parse(JSON.stringify(settingsModule.data.listOfMenu))
+
+    },
+    handleFeatureToggle(enabled, id) {
+      // alert("hii",id,enabled)
+      console.log("hello")
+      console.log("id-ena", id, enabled)
+      this.listOfExperiences = this.listOfExperiences.map(item => {
```

Comment:
If you are fetching the this.listOfExperiences from the store then what is the purpose of this ?

Reply:
Refactored code